docs/api/concepts.md

```diff
@@ -0,0 +1,82 @@
+<!-- front-matter
+id: api-concepts
+title: API Concepts
+hide_title: true
+sidebar_label: Concepts
+-->
+
+# Concepts
+
+The following concepts are prerequisites to understanding the API docs. They will be referenced throughout, refer back to this page for detailed explanations.
+
+If you're new here, begin with the [Getting Started Guide][quick-start-docs].
+
+## Vinyl
+
+Vinyl is a metadata object that describes a file. The main properties of a Vinyl instance are `path` and `contents` - core aspects of a file on your file system. Vinyl objects can be used to describe files from many sources - on a local file system or any remote storage option.
+
+## Vinyl adapters
+
+While Vinyl provides a way to describe a file, a way to access these files is needed. Each file source is accessed using a Vinyl adapter.
+
+An adapter exposes:
+* A method with the signature `src(globs, [options])` and returns a stream that produces Vinyl objects.
+* A method with the signature `dest(folder, [options])` and returns a stream that consumes Vinyl objects.
+* Any extra methods specific to their input/output medium - such as the `symlink` method `vinyl-fs` provides. They should always return streams that produce and/or consume Vinyl objects.
+
+## Tasks
+
+Each gulp task is an asynchronous JavaScript function that either accepts an error-first callback or returns a stream, promise, event emitter, child process, or observable. Due to some platform limitations, synchronous tasks aren't supported.
```

Comment:
Is this enough information for the "Tasks" definition?  It links out to "Creating Tasks" for more information